### PR TITLE
Distribution: Added BSD OS family

### LINKED ISSUE #43739: Missing OS families for DragonflyBSD, OpenBSD, NetBSD
##### SUMMARY

DragonflyBSD, OpenBSD, and NetBSD have distribution entries but lack family entries.

https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/facts/system/distribution.py

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
distribution

##### ANSIBLE VERSION
devel

### PR DESCRIPTION
**Pull request by anis016**
OS families added for Dragonfly, OpenBSD and NetBSD

SUMMARY
Added OS families Map for Dragonfly, OpenBSD and NetBSD. Fixes #43739

ISSUE TYPE
Bugfix Pull Request

### CODE REVIEW

**Comment by anis016** [comment]
cc @gundalow @samdoran

**Comment by Akasurde** [comment]
@anis016 Thanks for the PR. Could you please add a test in [this](https://github.com/ansible/ansible/blob/devel/test/units/module_utils/test_distribution_version.py) ?

needs_info

**Comment by anis016** [comment]
@Akasurde, I understand that I need to install individual OS and run something like this? https://github.com/ansible/ansible/issues/31501#issuecomment-340861535 Is this right? and also which version of OS should I use to generate the information?

File: lib/ansible/module_utils/facts/system/distribution.py [hunk 1]
```diff
@@ -483,7 +483,10 @@ class Distribution(object):
                      'HP-UX': ['HPUX'],
                      'Darwin': ['MacOSX'],
                      'FreeBSD': ['FreeBSD', 'TrueOS'],
-                     'ClearLinux': ['Clear Linux OS', 'Clear Linux Mix']}
+                     'ClearLinux': ['Clear Linux OS', 'Clear Linux Mix'],
+                     'DragonFly': ['DragonflyBSD', 'Gentoo/DragonflyBSD'],
+                     'OpenBSD': ['OpenBSD'],
+                     'NetBSD': ['NetBSD']}
```
Comment (samdoran) [inline lib/ansible/module_utils/facts/system/distribution.py]: Line 488 and 489 aren't needed unless you want to add additional distributions in those families.

Comment (anis016) [reply lib/ansible/module_utils/facts/system/distribution.py]: Alright! I will remove those. @samdoran is it okay if I chose the latest distribution (ClearLinux/DragonFly) for testing? I requested for info but still didn't get one!

Comment (samdoran) [reply lib/ansible/module_utils/facts/system/distribution.py]: Yes, using the latest distribution for testing is fine.

**Comment by anis016** [comment]
I am having problem running dragonflybsd OS for adding the testcases (the package issue as below while trying to install python) can someone please help with this to add the testcases?

```
Updating Avalon repository catalogue...
pkg: http://mirror-master.dragonflybsd.org/dports/dragonfly:5.0.2:x86:64/LATEST/meta.txz:
Not Found
repository Avalon has no meta file, using default settings
pkg: http://mirror-master.dragonflybsd.org/dports/dragonfly:5.0.2:x86:64/LATEST/packagesite.txz:
Not Found
Unable to update repository Avalon
```

**Comment by samdoran** [comment]
I have zero experience working with DragonFly BSD. Try reading out to the [DragonFly BSD community](https://www.dragonflybsd.org/mailinglists/) for help.

**Comment by anis016** [comment]
@dagwieers, @samdoran, @gundalow I am sorry, I am not able to move on with this ticket any further! I tried couple of days with the DragonFlyBSD but I couldn't figure out the internet connection from the OS and none helpful searches came out from the Google search. I would really like if someone can overtake this PR. Thank you!

**Comment by samdoran** [comment]
I'll work on moving this through.

**Comment by samdoran** [comment]
@anis016 I would appreciate feedback on the values now returned for DragonFly BSD:
Facts with this PR:
```
> ansible lab-dflybsd5 -m setup -a 'filter=ansible_distribution*'
lab-dflybsd5 | SUCCESS => {
    "ansible_facts": {
        "ansible_distribution": "DragonFly",
        "ansible_distribution_major_version": "5",
        "ansible_distribution_release": "5.6-RELEASE",
        "ansible_distribution_version": "5.6.2"
    },
    "
```

Currently what Ansible returns:
```
lab-dflybsd5 | SUCCESS => {
    "ansible_facts": {
        "ansible_distribution": "DragonFly",
        "ansible_distribution_release": "5.6-RELEASE",
        "ansible_distribution_version": "DragonFly v5.6.2-RELEASE #3: Sa"
    },
    "changed": false
}
```

I am mainly wondering about `ansible_distribution_release`. Should it be `5.6.2-RELEASE` or `5.6-RELEASE`? I'm not a BSD user so I'm not sure what is accurate and/or most useful.

**Comment by grayed** [comment]
@samdoran According to [source repo](http://gitweb.dragonflybsd.org/?p=dragonfly.git;a=summary), DragonFly BSD uses X.Y-RELEASE tags, like (most?) other BSDs do.

**Comment by samdoran** [comment]
@grayed Thank you for the feedback. I'm mainly wondering do BSD users find a need to know `5.6.2` or is `5.6` enough?

**Comment by Akasurde** [comment]
superseded by https://github.com/ansible/ansible/pull/70748

**Comment by Akasurde** [comment]
@anis016 Thanks for the contribution.

**Pull request closed without merging**